File: full-stack-asset-transfer-guide/applications/conga-cards/src/commands/read.ts
/*
 * Copyright contributors to the Hyperledgendary Full Stack Asset Transfer Guide project
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Gateway } from '@hyperledger/fabric-gateway';
import { CHAINCODE_NAME, CHANNEL_NAME } from '../config';
import { AssetTransfer } from '../contract';
import { assertDefined } from '../utils';

export default async function main(gateway: Gateway, args: string[]): Promise<void> {
    const assetId = assertDefined(args[0], 'Arguments: <assetId>');

    const network = gateway.getNetwork(CHANNEL_NAME);
    const contract = network.getContract(CHAINCODE_NAME);

    const smartContract = new AssetTransfer(contract);
    const asset = await smartContract.readAsset(assetId);

    const assetsJson = JSON.stringify(asset, undefined, 2);
    console.log(assetsJson);
}

File: full-stack-asset-transfer-guide/applications/trader-go/commands/read.go
package commands

import (
	"encoding/json"
	"fmt"

	"github.com/hyperledger/fabric-gateway/pkg/client"
)

// cmdRead reads and prints a single asset from the ledger.
// Arguments: <assetId>
func cmdRead(gw *client.Gateway, args []string) error {
	if len(args) < 1 {
		return fmt.Errorf("arguments: <assetId>")
	}

	network := gw.GetNetwork(channelName())
	contract := network.GetContract(chaincodeName())

	smartContract := NewAssetTransfer(contract)
	asset, err := smartContract.ReadAsset(args[0])
	if err != nil {
		return err
	}

	data, err := json.MarshalIndent(asset, "", "  ")
	if err != nil {
		return fmt.Errorf("failed to marshal asset: %w", err)
	}

	fmt.Println(string(data))
	return nil
}
